tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -79,27 +45,4 @@ private function assertIterationCallsFetchOncePerStep(Traversable $iterator, int
             $this->assertEquals($i + 1, $calls);
         }
     }
-
-    /**
-     * @return string[][]
-     */
-    public static function statementProvider() : iterable
```

Comment:
It is fine to remove these tests. They enforce the lazy behavior in all statement classes however it's achieved by just using the statement iterator. As long as the statement iterator is instantiated internally by the implementations, you cannot really enforce it by a test w/o complicating the API. I believe not having these tests is a reasonable tradeoff.